src/coreWebVitals.ts

```diff
@@ -0,0 +1,160 @@
+};
+
+const onReport: ReportHandler = (metric) => {
+	switch (metric.name) {
+		case 'FCP':
+			// Browser support: Chromium, Firefox, Safari Technology Preview
+			coreWebVitalsPayload.fcp = roundWithDecimals(metric.value);
+			break;
+		case 'CLS':
+			// Browser support: Chromium,
+			coreWebVitalsPayload.cls = roundWithDecimals(metric.value);
+			break;
+		case 'LCP':
+			// Browser support: Chromium
+			coreWebVitalsPayload.lcp = roundWithDecimals(metric.value);
+			break;
+		case 'FID':
+			// Browser support: Chromium, Firefox, Safari, Internet Explorer (with the polyfill)
+			coreWebVitalsPayload.fid = roundWithDecimals(metric.value);
+			break;
+		case 'TTFB':
+			// Browser support: Chromium, Firefox, Safari, Internet Explorer
+			coreWebVitalsPayload.ttfb = roundWithDecimals(metric.value);
+			break;
+	}
+};
+
+type InitCoreWebVitalsOptions = {
+	isDev: boolean;
+
+	browserId?: string | null;
+	pageViewId?: string | null;
+
+	sampling?: number;
+	bypassSampling?: boolean;
+};
+
+/**
+ * Initialise sending Core Web Vitals metrics to a logging endpoint.
+ *
+ * @param init - the initialisation options
+ * @param init.isDev - used to determine whether to use CODE or PROD endpoints.
+ * @param init.browserId - identifies the browser. Usually available via `getCookie('bwid')`. Defaults to `null`
+ * @param init.pageViewId - identifies the page view. Usually available on `guardian.ophan.pageViewId`. Defaults to `null`
+ *
+ * @param init.sampling - sampling rate for sending data. Defaults to `0.01`.
+ * @param init.bypassSampling - send data regardless of sampling rate. Defaults to `false`.
+ *
+ * @param metricsSentCallback - Optional callback, triggered after metrics are queued for sending
+ */
+export const initCoreWebVitals = (
+	{
+		browserId = null,
+		pageViewId = null,
+		bypassSampling = false,
+		sampling = 1 / 100, // 1% of page view by default
+		isDev,
+	}: InitCoreWebVitalsOptions,
+	metricsSentCallback?: (queued?: boolean) => void,
+): void => {
+	coreWebVitalsPayload.browser_id = browserId;
+	coreWebVitalsPayload.page_view_id = pageViewId;
+
+	if (!browserId || !pageViewId)
+		console.warn(
+			'browserId or pageViewId missing from Core Web Vitals.',
+			'Resulting data cannot be joined to page view tables',
+			{ browserId, pageViewId },
+		);
+
+	if (sampling < 0 || sampling > 1)
+		console.warn(
+			'Core Web Vitals sampling is outside the 0 to 1 range: ',
+			sampling,
+		);
+	if (sampling === 0) console.warn('Core Web Vitals are sampled at 0%');
+	if (sampling === 1) console.warn('Core Web Vitals are sampled at 100%');
+
+	// By default, sample a percentage of page views
+	const pageViewInSample = Math.random() < sampling;
+	// Unless we are forcing sending metrics for this page view
+	// via initialisation or calling bypassSampling()
+	if (bypassSampling || pageViewInSample) shouldSendMetrics = true;
+	// Or using a specific hash
+	if (window.location.hash === '#bypassSampling') shouldSendMetrics = true;
```

Comment:
`#bypassSampling` is a quite generic, how about this?

```suggestion
	if (window.location.hash === '#bypassCVWSampling') shouldSendMetrics = true;
```